### ISSUE TITLE
NullPointerException in StoreCached.flush()

### ISSUE THREAD

**Issue by bdupras** [jankotek/mapdb#660]
Ran into this intermittent NPE on `org.mapdb.DB.commit()`. We were calling `commit()` unnecessarily inside a tight single-threaded loop. Observed in a system running 2.0-beta10.

```
java.lang.NullPointerException 
        at org.mapdb.StoreCached.flush(StoreCached.java:380) 
        at org.mapdb.StoreDirect.commit(StoreDirect.java:1128) 
        at org.mapdb.Engine$CloseOnJVMShutdown.commit(Engine.java:487) 
        at org.mapdb.DB.commit(DB.java:2647) 
```

We've bumped to beta12 and removed the call to `commit()` from the tight loop. Will post here if we see again.

//cc @beala

**Comment by jankotek** [comment]
I believe it s caused by calling commit on already closed DB. In that case DB is not thread safe :(

Will fix in 3.0 branch.

**Comment by jankotek** [comment]
That is fixed in 3.0 branch
